tests/functional/adapter/materialization/test_materialized_view.py

```diff
@@ -280,3 +280,48 @@ def test_set_mv_properties(self, project):
         sql = f"SHOW CREATE MATERIALIZED VIEW {catalog}.{schema}.mat_view"
         results = run_sql_with_adapter(project.adapter, sql, fetch="all")
         assert "format = 'PARQUET'" in results[0][0]
+
+
+@pytest.mark.iceberg
+@pytest.mark.skip_profile("starburst_galaxy")
+class TestIcebergMaterializedViewWithGracePeriodProperties(TestIcebergMaterializedViewBase):
+    # Configuration in dbt_project.yml
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "name": "mv_test",
+            "models": {
+                "+materialized": "materialized_view",
+                "+properties": {"format": "'PARQUET'"},
+                "+grace_period": "INTERVAL '3' SECOND",
+            },
+        }
+
+    # Everything that goes in the "seeds" directory
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {
+            "seed.csv": seed_csv,
+        }
+
+    # Everything that goes in the "models" directory
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "mat_view.sql": model_sql,
+        }
+
+    def test_set_mv_properties(self, project):
+        catalog = project.adapter.config.credentials.database
+        schema = project.adapter.config.credentials.schema
+
+        # Seed seed
+        results = run_dbt(["seed"], expect_pass=True)
+        assert len(results) == 1
+
+        # Create MV
+        results = run_dbt(["run"], expect_pass=True)
+        assert len(results) == 1
```

Comment:
```suggestion
        results, logs = run_dbt_and_capture(["--debug", "run"], expect_pass=True)
        assert len(results) == 1
        assert "grace period INTERVAL '3' SECOND" in logs
```

Please add assertion to check if `grace period` is added to sql (you need to import `run_dbt_and_capture`)